Src/UnitTests/CoravelUnitTests/Events/EventTests.cs

```diff
@@ -34,6 +34,32 @@ public async Task TestRegisterOneListener()
             Assert.Equal(1, listenersExecutedCount);
         }
 
+        [Fact]
+        public async Task TestRegisterTwoListenerWithDifferentRegisterCall()
+        {
+            int listenersExecutedCount = 0;
+
+            var services = new ServiceCollection();
+            services.AddTransient<Action>(p => () => listenersExecutedCount++); // This is injected into the listeners via DI
+            services.AddEvents();
+            services.AddTransient<TestListener1ForEvent1>();
+            services.AddTransient<TestListener2ForEvent1>();
+            services.AddTransient<TestListenerForEvent2>();
+            var provider = services.BuildServiceProvider();
+
+            var dispatcher = provider.GetRequiredService<IDispatcher>() as Dispatcher;
+
+            dispatcher.Register<TestEvent1>()
+                .Subscribe<TestListener1ForEvent1>();
+
+            dispatcher.Register<TestEvent1>()
+                .Subscribe<TestListener2ForEvent1>();
+
+            await dispatcher.Broadcast(new TestEvent1());
+
+            Assert.Equal(2, listenersExecutedCount);
```

Comment:
Looks great! I have one suggestion. Why don't test to make sure that each specific listener is called once each Something like this:

```diff
+          int listener1ExecutedCount = 0;
+          int listener2ExecutedCount = 0;

            var services = new ServiceCollection();
            services.AddEvents();
+          services.AddTransient<TestListener1ForEvent1>(s => new TestListener1ForEvent1(() => listener1ExecutedCount++));
+          services.AddTransient<TestListener2ForEvent1>(s => new TestListener2ForEvent1(() => listener2ExecutedCount++));
            var provider = services.BuildServiceProvider();

            var dispatcher = provider.GetRequiredService<IDispatcher>() as Dispatcher;

            dispatcher.Register<TestEvent1>()
                .Subscribe<TestListener1ForEvent1>();

            dispatcher.Register<TestEvent1>()
                .Subscribe<TestListener2ForEvent1>();

            await dispatcher.Broadcast(new TestEvent1());

+          Assert.Equal(1, listener1ExecutedCount);
+          Assert.Equal(1, listener2ExecutedCount);
```

Reply:
Sure! Great idea!